python/ray/autoscaler/_private/gcp/node.py

```diff
@@ -684,6 +684,13 @@ def create_instance(
             # https://cloud.google.com/tpu/docs/users-guide-tpu-vm#create-curl
             config["networkConfig"]["enableExternalIps"] = True
 
+        # replace serviceAccounts with serviceAccount, and scopes with scope
+        # this is necessary for the head node to work
+        # see here: https://tpu.googleapis.com/$discovery/rest?version=v2alpha1
+        if "serviceAccounts" in config:
```

Comment:
I actually switched machines recently, and haven't set up ray in development mode. 

Trying to quickly hack on it (right now) led to some opaque errors when trying to run tests (missing imports) likely due to how i was actually editing the code in site-packages rather than in a proper `pip install -e .` style and master being ahead by enough... 

Also, i'm on a M1 mac, and it seems like that's not supported in the nightlies, which is described as the way to set up in python only development mode: 
https://docs.ray.io/en/latest/ray-contribute/development.html#building-ray-python-only
 
> At this time, Apple Silicon ray wheels are being published for releases only. As support stabilizes, nightly wheels will be published in the future.
https://docs.ray.io/en/latest/ray-overview/installation.html#m1-mac-apple-silicon-support

-- All this said, I think I am not going to fight all of the above and figure out how to install ray from source on M1 right now just to add this test. I'm obviously open to anyone else contributing and adding the test who has the environment set up.

I'll loop back around to this in a few weeks or so (depending on if I have free time) and will hack on getting a dev environment set up or set up in some quick linux VM to add it.